Objective-C/CBLDatabase.mm

```diff
@@ -327,6 +335,28 @@ - (BOOL) purgeDocumentWithID: (NSString*)documentID error: (NSError**)error {
     }
 }
 
+#pragma mark - Blob Save/Get
+
+- (BOOL) saveBlob: (CBLBlob*)blob error: (NSError**)error {
+    return [blob installInDatabase: self error: error];
+}
+
+- (CBLBlob*) getBlob: (NSDictionary*)properties {
+    if (!properties[kBlobDigestProperty] ||
+        ![properties[kBlobDigestProperty] isKindOfClass: [NSString class]] ||
+        !properties[kBlobTypeProperty] || ![properties[kBlobTypeProperty] isEqual: @"blob"] ||
+        (properties[kBlobContentTypeProperty] &&
+         ![properties[kBlobContentTypeProperty] isKindOfClass: [NSString class]]) ||
+        (properties[kBlobLengthProperty] &&
+         ![properties[kBlobLengthProperty] isKindOfClass: [NSNumber class]])) {
+        
+        [NSException raise: NSInvalidArgumentException
+                    format: @"Property dictionary is missing the digest or @type"];
```

Comment:
Could be "The given blob's metadata might be missing the digest / @type key or containing invalid values".